assets/js/theme/custom/wishlist-modal.js

```diff
@@ -0,0 +1,38 @@
+import modalFactory from '../global/modal'
+
+export default function wishlistModal() {
+    const selectors = {
+        wishlist: '.js-wishlist-add',
+    }
+
+    const text = {
+        title: 'Login or register to add to wishlist!',
+        message: 'Please login or register to add this item to your wishlist.',
+        login: 'Login',
+        createAccount: 'Create Account',
+    }
+
+    const wishlistButtons = document.querySelectorAll(selectors.wishlist)
+    wishlistButtons?.forEach((button) => {
+        const isCustomer = button.getAttribute('customer')
+        if (isCustomer !== 'null') return
+        button?.addEventListener('click', (ev) => {
+            ev.preventDefault()
+            const modal = modalFactory('.js-preview-modal')
+            if (modal) {
+                modal.open()
+
+                const contentHTML = `
```

Comment:
I'd suggest having two "add to wishlist buttons" (if you need to handle wishlists for guests and logged-in users) and displaying one of those buttons—but directly in HTML — based on the customer variable. It would be easier to handle multilingualism this way.